sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -260,11 +260,11 @@ private[sql] class DataFrameImpl protected[sql](
 
   override def take(n: Int): Array[Row] = head(n)
 
-  override def collect(): Array[Row] = queryExecution.executedPlan.executeCollect()
+  override def collect(): Array[Row] = rdd.collect()
 
   override def collectAsList(): java.util.List[Row] = java.util.Arrays.asList(rdd.collect() :_*)
 
-  override def count(): Long = groupBy().count().rdd.collect().head.getLong(0)
+  override def count(): Long = rdd.count()
```

Comment:
Are these changes correct?  Or are you removing the optimizations that we have in place for count and collects?